src/sql/src/plan/query.rs

```diff
@@ -3120,27 +3124,36 @@ pub fn resolve_func(
     bail!("function {}({}) does not exist", name, types.join(", "))
 }
 
-fn plan_is_null_expr<'a>(
+fn plan_is_expr<'a>(
     ecx: &ExprContext,
     inner: &'a Expr<Aug>,
+    construct: IsExprConstruct,
     not: bool,
 ) -> Result<HirScalarExpr, anyhow::Error> {
     // PostgreSQL can plan `NULL IS NULL` but not `$1 IS NULL`. This is at odds
     // with our type coercion rules, which treat `NULL` literals and
     // unconstrained parameters identically. Providing a type hint of string
     // means we wind up supporting both.
     let expr = plan_expr(ecx, inner)?.type_as_any(ecx)?;
-    let mut expr = HirScalarExpr::CallUnary {
-        func: UnaryFunc::IsNull(expr_func::IsNull),
+    let func = match construct {
+        IsExprConstruct::NULL => UnaryFunc::IsNull(expr_func::IsNull),
+        IsExprConstruct::TRUE => UnaryFunc::IsTrue(expr_func::IsTrue),
+        IsExprConstruct::FALSE => UnaryFunc::IsFalse(expr_func::IsFalse),
+        IsExprConstruct::UNKNOWN => UnaryFunc::IsUnknown(expr_func::IsUnknown),
```

Comment:
Here you'll want to use the `ecx` in this function to check the scalartype of `expr`; should look something like `matches!(ecx.scalar_type(&expr), ScalarType::Bool)`, and then you'll need to error if not. lmk if you want any help on the front end of that work.

Reply:
I think actually push the call to `type_as_any` into the `match`, and then change it to `type_as(..., ScalarType::Boolean)` for `IsExprConstruct::{True, False, Unknown}`, but leave it as `type_as_any` for `NULL` (to capture the difference between `IS NULL` and `IS UNKNOWN`).